models/articles.model.js

```diff
@@ -1,5 +1,43 @@
 const db = require("../db/connection");
 
+exports.fetchArticles = (topic) => {
+  if (topic) {
+    return db
+      .query(
+        `SELECT articles.author, title, articles.article_id, topic, articles.created_at, articles.votes, CAST (COUNT(comment_id) AS INT) AS comment_count 
+      FROM articles 
+      JOIN comments ON articles.article_id = comments.article_id WHERE topic = $1 GROUP BY articles.article_id ORDER BY article_id ASC;`,
+        [topic]
+      )
+      .then((data) => {
+        return db
+          .query(`SELECT EXISTS (SELECT 1 FROM articles WHERE topic = $1)`, [
+            topic,
+          ])
+          .then((response) => {
+            if (response.rows[0].exists === true) {
```

Comment:
What is .exists()? I had a little google and I could not find anything related to a use case here. We could just check if response.rows[0] is a truthy value. if (response.rows[0]) would suffice.

Reply:
Yeah I probably made this more complex than needed - my second db query (`SELECT EXISTS (SELECT 1 FROM articles WHERE topic = $1)` returns an exists key which will be true or false if the value is in the topic row